### ISSUE TITLE
Reduce example segfaults

### ISSUE THREAD

**Issue by TheLoneWolfling** [incrediblesound/huo#17]
```
10
ASAN:SIGSEGV
=================================================================
==7741==ERROR: AddressSanitizer: SEGV on unknown address 0xbebe007c (pc 0x0807f6bf bp 0xf5311480 sp 0xff3cf4b0 T0)
    #0 0x807f6be in reduce_array execution_functions/reduce.c:20
    #1 0x808d147 in execute /home/twotothe/afl/orig/huo/execute.c:38
    #2 0x808c127 in execute /home/twotothe/afl/orig/huo/execute.c:115
    #3 0x808dac3 in execute_defined_func /home/twotothe/afl/orig/huo/execute.c:129
    #4 0x808dac3 in execute /home/twotothe/afl/orig/huo/execute.c:96
    #5 0x808cee2 in execute /home/twotothe/afl/orig/huo/execute.c:99
    #6 0x8049a6a in main /home/twotothe/afl/orig/huo/huo.c:60
    #7 0xf7036516 in __libc_start_main (/lib/i386-linux-gnu/libc.so.6+0x18516)
    #8 0x804a87c  (/home/twotothe/afl/orig/huo/huoafl+0x804a87c)
```

**Comment by TheLoneWolfling** [comment]
Fixed in my eof_fix branch.

**Comment by TheLoneWolfling** [comment]
Fixed by #15
